src/sentry/integrations/slack/actions/notification.py

```diff
@@ -120,12 +129,71 @@ def send_notification(event: GroupEvent, futures: Sequence[RuleFuture]) -> None:
                     "attachments": json.dumps(attachments),
                 }
 
+            rule = rules[0] if rules else None
+            rule_to_use = self.rule if self.rule else rule
+            rule_id = rule_to_use.id if rule_to_use else None
+            rule_action_uuid = self.data.get("action_uuid", None)
+            if not rule_action_uuid:
+                # We are logging because this should never happen, all actions should have an uuid
+                # We can monitor for this, and create an alert if this ever appears
+                _default_logger.info(
+                    "integration.slack.actions: No action uuid",
+                    extra={
+                        "rule_id": rule_id,
+                        "notification_uuid": notification_uuid,
+                    },
+                )
+
+            new_notification_message_object = NewIssueAlertNotificationMessage(
+                rule_fire_history_id=self._rule_fire_history.id, rule_action_uuid=rule_action_uuid
+            )
+
+            # Only try to get the parent notification message if the organization is in the FF
+            # We need to search by rule action uuid and rule id, so only search if they exist
+            if (
+                features.has(
+                    "organizations:slack-thread-issue-alert", event.group.project.organization
+                )
+                and rule_action_uuid
+                and rule_id
```

Comment:
the send_notification method is super long, can we make it more modular. For example this check can be its own method which is used twice. Setting the thread_ts can be its own methods as well.

Reply:
`send_notification` is a function declared in a function, and this is a pattern we can see across all the integrations. I would also like to make it modular and easier to use, but i don't want to refactor in this PR and increase the blast radius. I can try to figure out how to make it safer in a follow up PR, with another FF (disabled would keep the function as is, enabled would utilize the new method which is broken up)

Reply:
I'm thinking maybe just adding it as a method of the class, and just being able to pass in the method itself, which would help break it up? I just don't know if self.func will be allowed or it has to be independent/pure function